fms/models/llama.py

```diff
@@ -218,21 +228,23 @@ def __init__(
         if self.config.p_dropout:
             self.dropout = nn.Dropout(self.config.p_dropout)
 
-        self.reset_params()
-
     def get_config(self) -> LLaMAConfig:
         return self.config
 
     @classmethod
     def from_config(cls, config: LLaMAConfig) -> "LLaMA":
         return cls(config)
 
-    def reset_params(self):
-        # Modules are self-initializing, we're just going to down-scale the final prediction head to be
-        # mixed-fan (inputs and gradients scale to the same inverse factors) if it isn't tied
-        self.shared.head.weight.data.normal_(
-            0, 1 / math.sqrt(math.sqrt(self.width * self.shared.vocab_size))
-        )
+    def reset_parameters(self):
+        # Call reset_parameters for relevant sub-layers
+        for m in self.modules():
+            if (
+                isinstance(m, MultiHeadAttention)
+                or isinstance(m, WordEmbedding)
+                or isinstance(m, GatedLinearUnit)
+                or isinstance(m, LayerNormParameterized)
```

Comment:
would it be better to just `hasattr(m, 'reset_parameters')` instead of listing them? We could end up adding more variants of these modules.

Also, for consistency, I think we need to update the other models too?

Reply:
Unfortunately we can't just use `hasattr(n, 'reset_parameters')` because PyTorch submodules (including `nn.Linear`) also have a `reset_parameters`. So we could end up calling our initializer, then accidentally overriding it with the PyTorch defaults for the submodules. 

We should update the other models, I'll propagate these changes to Roberta/GPTBigCode (though it's not fully clear to what degree this init scheme is suited toward these models in practice)

Reply:
Updated other model `reset_parameters` and updated description to reflect this

Reply:
ah, I think to do it the way I was suggesting it would need to be `.children()` instead of .`modules()` and then there'd need to be a similar `reset_parameters` method on LLaMABlock. Then it would only trickle through on the explicitly added modules and not flow into underlying linear modules. I guess this is fine too though.

Reply:
Oh I see, yeah makes sense